src/main/java/school/hei/haapi/authentication/AwsCognitoJwtIdToken.java

```diff
@@ -0,0 +1,79 @@
+package school.hei.haapi.authentication;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.proc.BadJOSEException;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.exception.NotAuthorizedException;
+import school.hei.haapi.service.UserService;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class AwsCognitoJwtIdToken implements Serializable {
+
+  private final JwtConfiguration jwtConfiguration;
+  private final UserService userService;
+  private final ConfigurableJWTProcessor<SecurityContext> configurableJwtProcessor;
+
+  public Authentication authenticate(HttpServletRequest request)
+      throws NotAuthorizedException, ParseException, BadJOSEException, JOSEException {
+    String idToken = request.getHeader(this.jwtConfiguration.getHttpHeader());
+    if (idToken != null) {
+      JWTClaimsSet claims =
+          this.configurableJwtProcessor.process(this.getBearerToken(idToken), null);
+      validateIssuer(claims);
+      verifyIfIdToken(claims);
+      String email = getEmail(claims);
+      String username = getUserNameFrom(claims);
+      school.hei.haapi.model.User dbUser = userService.findByEmail(email);
+      Collection<GrantedAuthority> authorities =
+          List.of(new SimpleGrantedAuthority(dbUser.getRole()));
+      if (username != null) {
+        User user = new User(username, "", authorities);
+        return new JwtAuthentication(user, authorities);
+      }
+    }
+    return null;
+  }
+
+  private String getUserNameFrom(JWTClaimsSet claims) {
+    return claims.getClaims().get(this.jwtConfiguration.getUserNameField()).toString();
+  }
+
+  private String getEmail(JWTClaimsSet claims) {
+    return claims.getClaims().get(this.jwtConfiguration.getEmailField()).toString();
+  }
+
+  private void verifyIfIdToken(JWTClaimsSet claims) throws NotAuthorizedException {
+    if (!claims.getIssuer().equals(this.jwtConfiguration.COGNITO_IDENTITY_POOL_URL_FORMAT)) {
```

Comment:
1. Use Yoda conditions https://en.wikipedia.org/wiki/Yoda_conditions
2. Use static import for JwtConfiguration.COGNITO_IDENTITY_POOL_URL_FORMAT so that you'll just use COGNITO_IDENTITY_POOL_URL_FORMAT here